src/ram_key_value_store/mod.rs

```diff
@@ -0,0 +1,103 @@
+use super::key_value_store::Key;
+use super::key_value_store::KeyValueStore;
+use super::key_value_store::Size;
+use core::cell::Cell;
+use core::mem;
+use core::mem::MaybeUninit;
+
+#[cfg(test)]
+mod test;
+
+pub struct RamKeyValueStoreElement {
+    pub key: Key,
+    pub size: Size,
+}
+
+impl RamKeyValueStoreElement {
+    pub fn new<T: Sized>(key: Key) -> Self {
+        Self {
+            key: key,
+            size: mem::size_of::<T>() as u8,
+        }
+    }
+}
+
+pub struct RamKeyValueStore<'a> {
+    ram: &'a mut [Cell<u8>],
```

Comment:
just a small question to clarify things for my sick brain that is 90% mucus, but is this a mutable borrow solely to prevent others from being able to borrow it? Or am I missing something else?

Reply:
Since it's an array of `Cell` the `mut` is probably not necessary. I'll play with it and try to get rid of it.

Reply:
We discussed this offline and making a mutable borrow is helpful since it prevents anyone else from touching this while the key value store is using it.